### DISCUSSION TITLE
Problem of a re-renderer with a default value when using the mobx

### DISCUSSION THREAD

**Discussion by aleksandr-z** [react-hook-form/react-hook-form#2939] [Q&A] [answered]
Hello my dear friend. Sorry for my English. Faced a problem when using mobx and defaultValues. After updating the store, the field values are lost. Here's an example https://codesandbox.io/s/divine-firefly-dr0q1?file=/src/App.js in this example we change the status (isLoading) in mobx strore. After we get empty fields. If remove 'defaultValues' from Controller it is work rigth, but we have warning in console. We cannot set defaultValues in useForm because we have many differencing big form.
Please help me. 
Thanks

**Comment by bluebill1049** [comment, marked as answer]
`useEffect(() => reset(value), [reset])` ?

**Reply by aleksandr-z** [reply]
Thanks man. I understood. It is working.
